---
title: "Analysis of human myeloid cell activation using ATAC sequencing"
subtitle: "Project #5986"
author: "Vincent van Hoef"
date: "2/28/2022"
output: 
  bookdown::html_document2: 
    toc: true
    toc_float: true
    self_contained: false
    css: ./assets/style.css
    includes:
      before_body: ./assets/logo.html
---

```{r, include = FALSE, dev = "svglite"}

knitr::opts_chunk$set(
  echo=FALSE,
  message=FALSE,
  warnings=FALSE,
  dev = "svglite"
)
```

This is an intermediate report for the analysis of project #5986. It includes the preprocessing, quality control and downstream analysis (i.e. differential expression and pathway enrichment analysis). More raw data is available but not yet included in this report.

# Data Description

The experiment consist of 2 patient-derived cell lines; each either knocked down for IRAK3 or control and followed during 4 timepoints: 0, 1, 4 and 24 hours. So, a total of 16 samples to be analyzed:

```{r tab, results='asis', dev = "svglite"}
tab <- "
| Name | Treatment | Time | Donor |
|------|:-----------:|:------:|:-----:|
| C0_R1 | Control | 0 | 5 |
| C0_R2 | Control | 0 | 6 |
| C1_R1 | Control | 1 | 5 |
| C1_R2 | Control | 1 | 6 |
| C4_R1 | Control | 4 | 5 |
| C4_R2 | Control | 4 | 6 |
| C24_R1 | Control | 24 | 5 |
| C24_R2 | Control | 24 | 6 |
| IK0_R1 | Knockdown | 0 | 5 |
| IK0_R2 | Knockdown | 0 | 6 |
| IK1_R1 | Knockdown | 1 | 5 |
| IK1_R2 | Knockdown | 1 | 6 |
| IK4_R1 | Knockdown | 4 | 5 |
| IK4_R2 | Knockdown | 4 | 6 |
| IK24_R1 | Knockdown | 24 | 5 |
| IK24_R2 | Knockdown | 24 | 6 |
"
cat(tab)
```

The donors will serve as replicates to allow for a statistical analysis. The biggest interest is comparing the knockdown of IRAK3 versus the control for each time point. Also of interest is the time-dependent chromatin accesibility in the control cells after stimulation, so comparing C1, C4 and C24 versus C0.

# Contract

A contract was set up with the following deliverables:

* Raw data transfer to Bianca project
* Run best practice preprocessing pipeline
* Pre- and post alignment quality control
* Peak calling
* Differential peak analysis + annotation
* Pathway enrichment
* Report with all results, tables and analysis code

# Preprocessing

The data is delivered as standard fastq files containing the raw, unmapped read data. The data preprocessing was done by running the nf-core atacseq pipeline. Briefly, the following steps were performed:

1. Raw read QC
2. Adapter trimming
3. Alignment (BWA)
4. Mark duplicates
5. Merge alignments from multiple libraries of the same sample
    + Re-mark duplicates
    + Filtering to remove:
        * reads mapping to mitochondrial DNA
        * reads mapping to blacklisted regions
        * reads that are marked as duplicates
        * reads that arent marked as primary alignments
        * reads that are unmapped
        * reads that map to multiple locations
        * reads containing > 4 mismatches
        * reads that are soft-clipped
        * reads that have an insert size > 2kb
        * reads that map to different chromosomes
        * reads that arent in FR orientation
        * reads where only one read of the pair fails the above criteria
    + Alignment-level QC and estimation of library complexity
    + Create normalised bigWig files scaled to 1 million mapped reads
    + Generate gene-body meta-profile from bigWig files
    + Calculate genome-wide enrichment
    + Call broad peaks (MACS2)
    + Annotate peaks relative to gene features
    + Create consensus peakset across all samples and create tabular file to aid in the filtering of the data
    + Count reads in consensus peaks (featureCounts)
    + Differential accessibility analysis, PCA and clustering
    + Generate ATAC-seq specific QC html report
6. Create IGV session file containing bigWig tracks, peaks and differential sites for data visualisation (IGV).
7. Present QC for raw read, alignment, peak-calling and differential accessibility results (ataqv, MultiQC, R)


# Quality Control

Two very useful quality control softwares were run on the data: multiQC and ataqv. These contain many high-quality plots; most of them are interactive and with relevant help sections to help interpretation. Most figures can be filtered and downloaded as needed. If the figures are too small to see, please right-click and "Open Frame in New Tab"

## MultiQC

MultiQC is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools. The LIB sections are done for each library separately - so for each technical replicate and each forward and reverse read collection (64 samples in total). MERGED LIB are for the 16 biological samples (so, technical replicates merged and looking at fragments after mapping the forward and reverse read pairs). The QC tools summarized here are:

- FastQC: raw quality measures on base and read level (before and after trimming adapters)
- Cutadapt: QC of trimming step
- SAMTools: mapping statistics
- Preseq: library complexity
- Picard: diverse statistics (of special interest here is the Insert size)
- DeepTools: fingerprint plot and read distribution profile
- MACS2: peak count and frip score
- HOMER: peak annotation proportion
- featureCounts: proportion of reads assigned to known genetic features
- DESEeq2: PCA and sample similarity


```{r, out.width="100%"}
knitr::include_url("../results/multiqc/broadPeak/multiqc_report.html", height = "800px")
```


## ataqv

ataqv makes some ATAC specific quality plots. Again, below is an interactive report showing these plots. Help for the specific plots can be found by clicking on the question mark in each figure.

```{r, out.width="100%"}
knitr::include_url("../results/bwa/mergedLibrary/ataqv/broadPeak/html/index.html", height = "800px")
```

:::note
To summarize, the quality of the data looks good to excellent. There are no outlier samples that should be removed before the downstream analysis nor seem there be to be systematic differences between the different donors, time points or treatment.
:::

Below, a few ATAC-specific quality plots are highlighted.

## Fragment size  {.tabset .tabset-pills}

Typically, a successful ATAC-seq experiment should generate a fragment size distribution plot with decreasing and periodical peaks corresponding to the nucleosome-free regions (NFR) (< 100bp) and mono-, di-, and tri-nucleosomes (~200, 400, 600bp, respectively). All samples have a fragment size distribution showing the expected peaks.

```{r}
headers <- gsub(".mLb.*", "", list.files(path="../results/bwa/mergedLibrary/picard_metrics/pdf/", pattern = "*insert_size_histogram.pdf", full.names = FALSE))
plots <- list.files(path="../results/bwa/mergedLibrary/picard_metrics/pdf", pattern = "*insert_size_histogram.pdf", full.names = TRUE)
names(plots) <- headers
```


```{r, results='asis'}
for(h in names(plots)) {
  cat('### ', h, ' {.tabset .unnumbered}', sep = "")
  cat('  \n')
  cat('  \n')
  cat('  \n')
  cat('![', h, '](', plots[[h]], '){width=640px height=640px}', sep="")
  cat('  \n')
  cat('  \n')
  cat('</center>')
}
```

## FRiP

The ENCODE data standards for ATAC-seq data state among other things that the fraction of reads in called peak regions (GRiP score) should be greater than 0.3, although values above 0.2 are acceptable. Almost all samples clear these thresholds or are at least very near to 0.2. 

```{r}
knitr::include_graphics("../results/bwa/mergedLibrary/frip.png")
```

## Number of peaks

Peaks were only called when present in both replicates; this to reduce most of the donor specific effects. In total, so combining all peaks found in both replicates across all samples, the number of peaks is 82704. Below you can see an UpSet plot of how these are distributed among the samples.

![](../results/bwa/mergedLibrary/macs/broadPeak/consensus/consensus_peaks.mLb.clN.boolean.intersect.plot.pdf){width=860px height=640px}



# Data Responsibility

- **NBIS & Uppnex** Unfortunately, we do not have resources to keep any files associated with the support request. We suggest that you safely store the results delivered by us. In addition, we ask that you remove the files from UPPMAX/UPPNEX after analysis is completed. The main storage at UPPNEX is optimized for high-speed and parallel access, which makes it expensive and not the right place for long time archiving.
- **Sensitive data** Please note that special considerations may apply to the human-derived sensitive personal data. These should be handled according to specific laws and regulations.
- **Long-term backup** The responsibility for data archiving lies with universities and we recommend asking your local IT for support with long-term data archiving. Also the newly established Data Office at SciLifeLab may be of help to discuss other options.

# Acknowledgements

If you are presenting the results in a paper, at a workshop or conference, we kindly ask you to acknowledge us.

- **NBIS Staff** are encouraged to be co-authors when this is merited in accordance to the ethical recommendations for authorship, e.g. [ICMJE recommendations](http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html). If applicable, please include **Vincent van Hoef, National Bioinformatics Infrastructure Sweden, Science for Life Laboratory, Uppsala University**, as co-author. In other cases, NBIS would be grateful if support by us is acknowledged in publications according to this example: ["Support by NBIS (National Bioinformatics Infrastructure Sweden) is gratefully acknowledged"](https://www.nbis.se/resources/support.html).

- **UPPMAX** If your project has used HPC resources we kindly asks you to acknowledge UPPMAX and SNIC. If applicable, please add: ["The computations were performed on resources provided by SNIC through Uppsala Multidisciplinary Center for Advanced Computational Science (UPPMAX) under Project SNIC XXXX/Y-ZZZ"](https://www.uppmax.uu.se/support/faq/general-miscellaneous-faq/acknowledging-uppmax--snic--and-uppnex/).

- **NGI** In publications based on data from NGI Sweden, the authors must acknowledge SciLifeLab, NGI and UPPMAX: ["The authors would like to acknowledge support from Science for Life Laboratory, the National Genomics Infrastructure, NGI, and Uppmax for providing assistance in massive parallel sequencing and computational infrastructure"](https://ngisweden.scilifelab.se/info/faq#how-do-i-acknowledge-ngi-in-my-publication).

# Closing Procedures

You should soon be contacted by one of our managers, [Jessica Lindvall](jessica.lindvall@nbis.se) or [Henrik Lantz](henrik.lantz@nbis.se), with a request to close down the project in our internal system and for invoicing matters. If we do not hear from you within **30 days** the project will be automatically closed and invoice sent. Again, we would like to remind you about data responsibility and acknowledgements, see Data Responsibility and Acknowledgments sections.

You are naturally more than welcome to come back to us with further data analysis request at any time via [NBIS support](http://nbis.se/support/support.html).

Thank you for using NBIS and all the best for future research.